docs/module-definition.md

```diff
@@ -0,0 +1,55 @@
+|-----------------------|-----------------|---------------------------------------------------------------------------------------------------------------------------|
+| `field`               | string          | key to store result for project definition                                                                                |
+| `label`               | string          | displayed name for the prompt                                                                                             |
+| `options`             | list(string)    | A list of values for users to pick from                                                                                   |
+| `default`             | string          | Defaults to this value during prompt                                                                                      |
+| `value`               | string          | Skips prompt entirely when set                                                                                            |
+| `info`                | string          | Displays during prompt as extra information guiding user's input                                                          |
+| `fieldValidation`     | Validation      | Validations for the prompt value                                                                                          |
+| `type`                | enum(string)    | Built in custom prompts: currently supports [`AWSProfilePicker`]                                                            |
+| `execute`             | string          | executes commands and takes stdout as prompt result                                                                       |
+| `omitFromProjectFile` | bool            | Field is skipped from adding to project definition                                                                        |
+| `conditions`          | list(Condition) | Conditions for prompt to run, if supplied all conditions must pass                                                        |
+| `envVarName`          | string          | During `zero apply` parameters are available as env-vars, defaults to field name but can be overwritten with `envVarName` |
+
+### Condition
+| Parameters   | Type         | Description                                                       |
+|--------------|--------------|-------------------------------------------------------------------|
+| `action`     | enum(string) | type of condition, currently supports [`KeyMatchCondition`]         |
+| `matchField` | string       | Allows you to condition prompt based on another parameter's value |
+| `WhenValue`  | string       | Matches for this value to satisfy the condition                   |
+| `data`       | list(string) | Supply extra data for condition to run                            |
+
+
+### Validation
```

Comment:
Missing info on some of the other things like the `contitions` and `template` fields related to templating.

Reply:
definitely good call to add this documentation, I looked at conditions and thought oh its there, but theres actually 2 types

Reply:
Yeah the naming might be a bit confusing..